agent/functions/guard_test.go

```diff
@@ -0,0 +1,75 @@
+package functions
+
+import (
+	"strings"
+	"testing"
+)
+
+func TestGuardPath(t *testing.T) {
+	tests := []struct {
+		name    string
+		path    string
+		wantErr bool
+		errMsg  string
+	}{
+		{
+			name:    "正常系: 通常のパス",
+			path:    "path/to/file.txt",
+			wantErr: false,
+			errMsg:  "",
+		},
+		{
+			name:    "正常系: 空のパス",
+			path:    "",
+			wantErr: false,
+			errMsg:  "",
+		},
+		{
+			name:    "異常系: 親ディレクトリ参照",
+			path:    "../path/to/file",
+			wantErr: true,
+			errMsg:  "contains not allowed '..'",
+		},
+		{
+			name:    "異常系: チルダ使用",
+			path:    "~/path/to/file",
+			wantErr: true,
+			errMsg:  "contains not allowed '~'",
+		},
+		{
+			name:    "異常系: 重複スラッシュ",
+			path:    "path//to/file",
+			wantErr: true,
+			errMsg:  "contains not allowed '//'",
+		},
+		{
+			name:    "異常系: ルートパス開始",
+			path:    "/path/to/file",
```

Comment:
from Test Security Reviewer
Windows形式のパス区切り文字を使用したパストラバーサルの検証を追加します。これはクロスプラットフォームでの脆弱性を防ぐために重要です。

```suggestion
		{
			name:    "異常系: Windows形式の親ディレクトリ参照",
			path:    "..\\path\\to\\file",
			wantErr: true,
			errMsg:  "contains not allowed '..'",,
		},
```